lib/app/client.js

```diff
@@ -384,8 +386,30 @@ async function render (to, from, next) {
           promises.push(p)
       }
 
-      return Promise.all(promises)
-    }))
+      return allPromisesHandled(promises).catch(async (errors) => 
+        Promise.reject(
+          errors.length === 1 ?
```

Comment:
The logic inside Promise.reject is repeated multiply times (with minor argument differences.) I suggest refactor them into `compactMultiplyErrors`

Reply:
That's something I can change. I just initially didn't want `compactMultipleErrors` to make the assumption that the author always wanted it to return the first element in the array if it was the only one; plus I didn't think I repeated it enough to warrant DRY-ing it. I'll refactor, if you still want me to, in a new commit.